app/serializers/api/personalized_questionnaire_serializer.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Api
+  class PersonalizedQuestionnaireSerializer < ActiveModel::Serializer
+    attributes :uuid
+    attributes :questionnaire_title
+    attributes :questionnaire_content
+
+    def questionnaire_content
+      questionnaire[:content]
+    end
+
+    def questionnaire_title
+      questionnaire[:title]
+    end
+
+    private
+
+    def questionnaire
+      return @questionnaire if @questionnaire.present?
+      questionnaire = object.measurement.questionnaire
+      @questionnaire = QuestionnaireGenerator.generate_hash_questionnaire(
+        object.id,
+        questionnaire.content,
+        questionnaire.name
```

Comment:
Volgens mij moet dit `questionnaire.title` zijn ipv `questionnaire.name`. Zie de andere plaatsen waar we de questionnaire_generator aanroepen.

Het idee was dat `questionnaire.title` getoond zou worden in een grote h1 tag bovenaan de vragenlijst, maar later bleek dat Mandy enzo dat niet zo mooi vonden, dus nu hebben geloof ik bijna alle vragenlijsten een blank title (lege string), maar dat is wel het veld dat variabelen mag bevatten en geevalueerd wordt.

`questionnaire.name` was bedoeld als interne naam, voor onszelf, die we eigenlijk niet aan de buitenwereld willen tonen. Dit attribuut is ook uniek (titles zijn dat uiteraard niet), zodat je een questionnaire op name kan identificeren. Het was bedoeld als een soort van interne beschrijving van de vragenlijst.

`questionnaire.key` hadden we er later bijgevoegd (in de rol van het `quby_key` attribuut) omdat `questionnaire.name` toch niet restrictief genoeg was, en we eigenlijk geen spaties en andere rare symbolen in de questionnaire keys willen als we ze bijv opvragen via de API. Ook zijn de questionnaire keys dus letterlijk de bestandsnamen waar de questionnaires in opgeslagen zijn. Questionnaire keys zijn op dit moment niet uniek in de database maar volgens mij zou dat eigenlijk wel moeten (ook al kun je twee bestanden met dezelfde naam in verschillende subdirectories van de questionnaire seeds hebben: dat moeten we dan gewoon niet doen).



Samenvattend:

- Questionnaire Title: wat getoond wordt aan de gebruiker bovenaan de vragenlijst in een grote h1 tag. Mag leeg zijn en is dat ook vaak. Hier kunnen ook {{variabelen}} inzitten die dan geevalueerd worden.
- Questionnaire Name: een interne unieke beschrijving van de vragenlijst voor onszelf (niet voor buitenaf). De questionnaire names worden bijvoorbeeld gebruikt in de dropdown in de admin interface voor het previewen van een questionnaire. Een name kan hier dus iets descriptiever zijn dan een key waar geen spaties en rare tekens in mogen staan.
- Questionnaire Key: De bestandsnaam waarin een questionnaire wordt opgeslagen, mag geen spaties of rare tekens bevatten (alleen letters, cijfers, en underscores). Wordt ook gebruikt in de API voor het identificeren van een vragenlijst geloof ik.

Reply:
check oke, ik dacht dat title eigenlijk niet gebruikt werd. thx